ext/-test-/asan/asan.c

```diff
@@ -0,0 +1,28 @@
+#include "ruby/ruby.h"
+
+static VALUE
+asan_enabled_p(VALUE self)
+{
+#if defined(__SANITIZE_ADDRESS)
+    /* GCC defines this when asan is enabled */
+    return Qtrue;
+#elif defined(__has_feature)
+    /* clang uses __has_feature for determining asan */
+# if __has_feature(address_sanitizer)
+    return Qtrue;
+# else
+    return Qfalse;
+# endif
```

Comment:
`__has_feature` is expanded to a compile-time constant, not a preprocessor constant.
So you can write as:

```suggestion
    return __has_feature(address_sanitizer) ? Qtrue : Qfalse;
```

Reply:
Oh that's a good idea, I've done that.